mirrord/layer/src/file/ops.rs

```diff
@@ -850,6 +851,37 @@ pub(crate) fn rename(old_path: Detour<PathBuf>, new_path: Detour<PathBuf>) -> De
     })??)
 }
 
+pub(crate) fn ftruncate(fd: RawFd, length: i64) -> Detour<()> {
+    let fd = get_remote_fd(fd)?;
+    Detour::Success(common::make_proxy_request_with_response(
+        FtruncateRequest { fd, length },
+    )??)
+}
+
+pub(crate) fn futimens(fd: RawFd, times: Option<[Timespec; 2]>) -> Detour<()> {
+    let fd = get_remote_fd(fd)?;
+    Detour::Success(common::make_proxy_request_with_response(
+        FutimensRequest { fd, times },
+    )??)
+}
+
+pub(crate) fn fchown(fd: RawFd, owner: u32, group: u32) -> Detour<()> {
```

Comment:
The only thing I am not sure about is that the source of owner and group ids may not be acquired by calling stat/fstat functions. And we end up setting local owner/group ids on the remote file. It is unlikely to succeed so we fallback to bypass and call it locally. But if it succeeds, :monkey-lookaway:. Is this a valid concern? Correct me if I am wrong please.